scene/main/scene_tree.cpp

```diff
@@ -1539,6 +1539,15 @@ SceneTree::SceneTree() {
 #ifdef TOOLS_ENABLED
 	edited_scene_root = nullptr;
 #endif
+
+#ifdef DEBUG_ENABLED
+	if (Engine::get_singleton()->is_editor_hint()) {
+		// enable NavigationServer3D when in Editor or else navmesh edge connections are invisible
+		// on runtime tests SceneTree has "Visible Navigation" set and main iteration takes care of this
+		NavigationServer3D::get_singleton()->set_active(true);
+		NavigationServer3D::get_singleton_mut()->set_debug_enabled(true);
+	}
+#endif // DEBUG_ENABLED
 }
```

Comment:
Does this have to be in `SceneTree`? i.e. can it be kept local to the `NavigationServer`?

Reply:
Yeah right, I guess I added it here because the SceneTree/Main where previously the one disabling it in the Editor but makes more sense to have it on the server.